packages/contracts/contracts/UnirepApp.sol

```diff
@@ -31,9 +37,17 @@ contract UnirepApp {
     // sign up users in this app
     function userSignUp(
         uint256[] memory publicSignals,
-        uint256[8] memory proof
+        uint256[8] memory proof,
+        uint256 hashUserId
```

Comment:
actually the Unirep.sol will handle the double signup

https://github.com/Unirep/Unirep/blob/a55e9653f85760e95cba664926ed2ae2e3007577/packages/contracts/contracts/Unirep.sol#L159-L160

and one of the public signals is actually identity commitment (`hash(idSecret)`)

Reply:
Our sign up flow is: 
frontend get user id -> hash -> wallet sign -> generate identity commitment
Saving hashedUserId in contract is to prevent user using multiple wallet account signing single twitterid to sign up.